### ISSUE TITLE
Material Fail on AMF importer with 3DS MAX 2021

### ISSUE THREAD

**Issue by Baconator650** [Gravemind2401/Reclaimer#13]
![image](https://user-images.githubusercontent.com/78010776/128109154-50abc1dd-2a7d-4b58-86c1-be2c1df31185.png)

**Comment by choel** [comment]
3DS Max 2019 is now the oldest version readily available, and also the first version that doesn't work with this script: https://knowledge.autodesk.com/customer-service/account-management/users-software/previous-versions/eligible-previous-version-subscription. 

Do you have any interest in updating the script, with the primary road block being the lack of license? If so, I would be interested in providing the license to facilitate updating the script. I could invite you to my Autodesk team ( https://knowledge.autodesk.com/customer-service/account-management/users-software/user-management/manage-teams-autodesk-account ) and assign you my indie license, or purchase another indie license.

**Comment by Gravemind2401** [comment]
I've since been able to get more information on this and the issue is the Arnold renderer in the newer versions of Max. The geometry should import just fine, including material IDs and UVs, but the automatic material setup is not compatible with because Arnold doesn't support the legacy scanline materials (though it does have a compatibility setting to enable them with side effects). The material setup will work just fine if you change your settings to use the scanline renderer instead, but of course this limits you to the legacy material types and rendering setup.
Unfortunately this means the "fix" would be to make a whole new material setup script for the new material types, including finding an appropriate and simple way to do Halo-like materials in Arnold. Since I've been wanting to replace AMF for a while now anyway, its probably worth just supporting it with a new format rather than trying to make it work in the existing script. I hope to start doing more work on Reclaimer again soon, though I have some other things to sort out first and progress has been slow.

**Comment by choel** [comment]
Thank you for the updated information! It is appreciated. 

For the next person that comes along with this issue, here is a workaround I found;
1. Export the desired asset from Reclaimer in the AMF format.
2. Import the AMF into Blender using the AMF script with the scale set to "Halo".
3. Export from Blender in FBX format
4. Import into #DS MAx from FBX with the scale set to "Centimeters" or 1.0
5. Delete root "Armature" object and then reparent geometry to the next root object.

I've tested this out mostly on weapons in H2 with no issues -- haven't tried it with H3 objects or Bipeds yet.
